parallaxpager/src/main/java/com/prolificinteractive/parallaxpager/ParallaxLayoutInflater.java

```diff
@@ -1,28 +1,290 @@
 package com.prolificinteractive.parallaxpager;
 
+import android.annotation.TargetApi;
 import android.content.Context;
+import android.os.Build;
+import android.util.AttributeSet;
 import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import org.xmlpull.v1.XmlPullParser;
 
-public class ParallaxLayoutInflater extends LayoutInflater {
+/**
+ * Custom LayoutInflater that intercepts View creation and notifies ParallaxFactory of new views.
+ * This class is a modified version from the Calligraphy project
+ * @see uk.co.chrisjenx.calligraphy.CalligraphyLayoutInflater
+ */
+class ParallaxLayoutInflater extends LayoutInflater {
 
-  protected ParallaxLayoutInflater(Context context) {
-    super(context);
-    setUpLayoutFactory();
-  }
+  private static final String[] sClassPrefixList = {
+      "android.widget.",
+      "android.webkit."
+  };
+
+  private final ParallaxFactory mParallaxFactory;
+  // Reflection Hax
+  private boolean mSetPrivateFactory = false;
+  private Field mConstructorArgs = null;
 
-  protected ParallaxLayoutInflater(LayoutInflater original, Context newContext) {
+  protected ParallaxLayoutInflater(LayoutInflater original, Context newContext,
+      ParallaxFactory factory) {
     super(original, newContext);
-    setUpLayoutFactory();
+    this.mParallaxFactory = factory;
+    setUpLayoutFactories();
+  }
+
+  @Override
+  public LayoutInflater cloneInContext(Context newContext) {
+    return new ParallaxLayoutInflater(this, newContext, mParallaxFactory);
   }
 
-  private void setUpLayoutFactory() {
-    if (!(getFactory() instanceof ParallaxFactory)) {
-      setFactory(new ParallaxFactory(getFactory()));
+  // ===
+  // Wrapping goodies
+  // ===
+
+  @Override
+  public View inflate(XmlPullParser parser, ViewGroup root, boolean attachToRoot) {
+    setPrivateFactoryInternal();
+    return super.inflate(parser, root, attachToRoot);
+  }
+
+  /**
+   * We don't want to unnecessary create/set our factories if there are none there. We try to be
+   * as lazy as possible.
+   */
+  private void setUpLayoutFactories() {
+    // If we are HC+ we get and set Factory2 otherwise we just wrap Factory1
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
```

Comment:
This line was taken from Calligraphy, which supports sdk 7. Right now we need sdk 14 because of the animations, but we can keep this to fix issue #1.